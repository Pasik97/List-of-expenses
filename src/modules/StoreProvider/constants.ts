import { ReactNode } from "react";
import Stores from "stores/Stores";

export interface StoreProviderProps {
   store: Stores;
   children: ReactNode;
}
