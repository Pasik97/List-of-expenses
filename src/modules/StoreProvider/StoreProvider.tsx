import React, { createContext, } from 'react';
import { StoreProviderProps } from './constants';
import Stores from 'stores/Stores';

export const StoreProviderContext = createContext<Stores>({} as Stores);

const StoreProvider: React.FC<StoreProviderProps> = ({
   children,
   store,
}) => (
   <StoreProviderContext.Provider value={store}>
      {children}
   </StoreProviderContext.Provider>
);

export default StoreProvider;
