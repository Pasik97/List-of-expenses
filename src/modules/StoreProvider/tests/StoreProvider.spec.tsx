import { create } from 'react-test-renderer';
import { mockStore } from './mockData';
import StoreProvider from '../StoreProvider';

describe('<StoreProvider />', () => {
   it('should render div wrapper in provider', () => {
      const component = create(<StoreProvider store={mockStore}><div /></StoreProvider>);

      expect(component.toJSON()).toMatchSnapshot();
   });
});
