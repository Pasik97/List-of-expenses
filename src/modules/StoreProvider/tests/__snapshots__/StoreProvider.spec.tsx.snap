// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreProvider /> should render div wrapper in provider 1`] = `<div />`;
